### PR TITLE
Bump Elastic Stack version to 7.16.1

### PR DESCRIPTION
**Pull request by david-kow**

### CODE REVIEW

**Comment by david-kow** [comment]
run full pr build

File: config/recipes/elastic-agent/fleet-custom-logs-integration.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: Kibana
 metadata:
```
Comment (thbkrkr) [inline config/recipes/elastic-agent/fleet-custom-logs-integration.yaml:4]: Nit: remove the comment located on the first line of the file.

**Comment by thbkrkr** [comment]
Full PR build failure:
```
=== RUN   TestFleetKubernetesIntegrationRecipe/ES_data_should_pass_validations
Retries (30m0s timeout): .................................................................................................................
    step.go:43: 
        	Error Trace:	utils.go:87
        	Error:      	Received unexpected error:
        	            	elasticsearch client failed for https://elasticsearch-gs7d-es-http.e2e-hextv-mercury.svc:9200/_data_stream/logs-elastic_agent-default:
        	            	404 Not Found: {Status:404 Error:{CausedBy:{Reason: Type:} 
        	            	Reason:no such index [logs-elastic_agent-default] 
        	            	Type:index_not_found_exception RootCause:[{Reason:no such index [logs-elastic_agent-default]
        	            	Type:index_not_found_exception}]}}
```

Looks similar to #4878 fixed by #4879.

**Comment by david-kow** [comment]
run full pr build

**Comment by david-kow** [comment]
run full pr build

**Comment by david-kow** [comment]
run full pr build

**Comment by david-kow** [comment]
run full pr build

**Review by pebrc — APPROVED** [review]
LGTM

**Comment by david-kow** [comment]
run full pr build

The previous one failed on license check for maps (?).

**Comment by david-kow** [comment]
We failed on Fleet tests this time. After investigating this for a bit, it seems we are hitting a known issue: https://www.elastic.co/guide/en/fleet/current/release-notes-7.16.0.html#known-issues-7.16.0. I'm testing locally after reverting to `7.16.1` right now.

**Comment by david-kow** [comment]
run full pr build

**Pull request merged**